extension/oidcextension/authenticator.go

```diff
@@ -29,47 +29,51 @@ import (
 
 	"github.com/coreos/go-oidc"
 	"google.golang.org/grpc"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/configauth"
 )
 
 type oidcAuthenticator struct {
 	attribute string
-	config    OIDC
+	config    Config
 	provider  *oidc.Provider
 	verifier  *oidc.IDTokenVerifier
 
-	unaryInterceptor  unaryInterceptorFunc
-	streamInterceptor streamInterceptorFunc
+	unaryInterceptor  configauth.UnaryInterceptorFunc
+	streamInterceptor configauth.StreamInterceptorFunc
+	host              component.Host
 }
 
 var (
-	_ Authenticator = (*oidcAuthenticator)(nil)
-
-	errNoClientIDProvided                = errors.New("no ClientID provided for the OIDC configuration")
-	errNoIssuerURL                       = errors.New("no IssuerURL provided for the OIDC configuration")
+	errNoAudienceProvided                = errors.New("no Audience provided for the Config configuration")
+	errNoIssuerURL                       = errors.New("no IssuerURL provided for the Config configuration")
 	errInvalidAuthenticationHeaderFormat = errors.New("invalid authorization header format")
-	errFailedToObtainClaimsFromToken     = errors.New("failed to get the subject from the token issued by the OIDC provider")
-	errClaimNotFound                     = errors.New("username claim from the OIDC configuration not found on the token returned by the OIDC provider")
-	errUsernameNotString                 = errors.New("the username returned by the OIDC provider isn't a regular string")
-	errGroupsClaimNotFound               = errors.New("groups claim from the OIDC configuration not found on the token returned by the OIDC provider")
+	errFailedToObtainClaimsFromToken     = errors.New("failed to get the subject from the token issued by the Config provider")
```

Comment:
Are those changes intentional? Looks like the refactoring was too greedy :-)

Reply:
lol I think my refactor change name got a little too excited